src/client/app/components/RouteComponent.jsx

```diff
@@ -5,50 +5,69 @@
 import React from 'react';
 import { Router, Route, browserHistory } from 'react-router';
 import axios from 'axios';
-import HomeComponent from '../components/HomeComponent';
-import LoginComponent from './LoginComponent';
+import _ from 'lodash';
+import NotificationSystem from 'react-notification-system';
+import HomeComponent from './HomeComponent';
+import LoginContainer from '../containers/LoginContainer';
 import AdminComponent from './AdminComponent';
 import NotFoundComponent from './NotFoundComponent';
 
-/**
- * Middleware function that requires proper authentication for a page route
- * @param nextState The next state of the router
- * @param replace Function that allows a route redirect
- */
-function requireAuth(nextState, replace) {
-	function redirectRoute() {
-		replace({
-			pathname: '/login',
-			state: { nextPathname: nextState.location.pathname }
-		});
+export default class RouteComponent extends React.Component {
+	constructor(props) {
+		super(props);
+		this.requireAuth = this.requireAuth.bind(this);
 	}
-	const token = localStorage.getItem('token');
-	// Redirect route to login page if the auth token does not exist
-	if (!token) {
-		redirectRoute();
-		return;
+
+	componentWillReceiveProps(nextProps) {
+		if (!_.isEmpty(nextProps.notification)) {
+			this.notificationSystem.addNotification(nextProps.notification);
```

Comment:
Will this display only one notification at a time? Is there a way to display multiple? For example, if two API requests are made and both fail within a short time, it would be useful to show two error messages, because the first might be more important than the second (bad auth vs. a network timeout, for example).

Reply:
I believe that `componentWillReceiveProps` will be called twice for two actions, so it is fine.